### ISSUE TITLE
MergeOne-Regular.ttf: AssertionError on glyph entries when saving hotfixed TTF

### ISSUE THREAD

**Issue by felipesanches** [googlefonts/fontbakery#816]
When saving the final hotfixed TTF file for github_google/fonts/ofl/mergeone/MergeOne-Regular.ttf
FontBakery crashes with an AssertionError related to the glyphOrder/glyphs entries. This may mean some mistake in the way we hotfix these entries. I'll take a look at this one tomorrow morning. First step will be identifying exactly which check has hotfix routines that cause the bad data and then I'll fix it.

```
Traceback (most recent call last):
  File "./fontbakery-check-ttf.py", line 2685, in <module>
    main()
  File "./fontbakery-check-ttf.py", line 2677, in main
    font.save(font_file_output)
  File "/home/felipe/devel/github_felipesanches/scope-one/src/fonttools/Lib/fontTools/ttLib/__init__.py", line 202, in save
    self._writeTable(tag, writer, done)
  File "/home/felipe/devel/github_felipesanches/scope-one/src/fonttools/Lib/fontTools/ttLib/__init__.py", line 628, in _writeTable
    self._writeTable(masterTable, writer, done)
  File "/home/felipe/devel/github_felipesanches/scope-one/src/fonttools/Lib/fontTools/ttLib/__init__.py", line 631, in _writeTable
    tabledata = self.getTableData(tag)
  File "/home/felipe/devel/github_felipesanches/scope-one/src/fonttools/Lib/fontTools/ttLib/__init__.py", line 644, in getTableData
    return self.tables[tag].compile(self)
  File "/home/felipe/devel/github_felipesanches/scope-one/src/fonttools/Lib/fontTools/ttLib/tables/_m_a_x_p.py", line 45, in compile
    self.recalc(ttFont)
  File "/home/felipe/devel/github_felipesanches/scope-one/src/fonttools/Lib/fontTools/ttLib/tables/_m_a_x_p.py", line 64, in recalc
    self.numGlyphs = len(glyfTable)
  File "/home/felipe/devel/github_felipesanches/scope-one/src/fonttools/Lib/fontTools/ttLib/tables/_g_l_y_f.py", line 195, in __len__
    assert len(self.glyphOrder) == len(self.glyphs)
AssertionError
```

**Comment by felipesanches** [comment]
I just figured out that this is triggered by the hotfix of the following check:
"Font contains glyphs for whitespace characters?"

**Comment by davelab6** [comment]
Forget hotfixing, I don't use them, I report problem upstream

On 28 June 2016 at 14:58, Felipe Corrêa da Silva Sanches <
notifications@github.com> wrote:

> I just figured out that this is triggered by the hotfix of the following
> check:
> "Font contains glyphs for whitespace characters?"
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/googlefonts/fontbakery/issues/816#issuecomment-229148253>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAP9y1ehKgCHJScmmuy3rqVG-SLhogvdks5qQW7ugaJpZM4I_n-f>
> .
>



-- 
Cheers
Dave

**Comment by felipesanches** [comment]
ok, good

**Comment by felipesanches** [comment]
MergeOne is temporarily blacklisted in the "Font contains glyphs for whitespace characters?" check.

**Comment by felipesanches** [comment]
fixed at 05e6ee82ef7d0f48fca1c169cf808c72c5bae480
